### PR TITLE
ttd bid adapter: configurable endpoint 

### PR DESCRIPTION
**Pull request by tongwu-sh**
## Type of change
- [x] Feature

## Description of change

We want to allow users to specify their preferred TTD endpoint instead of the default endpoint for bidding.

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page. -->

### CODE REVIEW

**Review by patmmccann — CHANGES_REQUESTED** [review]
Couple of minor requests

**Comment by tongwu-sh** [comment]
> Hi @tongwu-sh ? It seems your adapter needs a tiny bit of additional maintenance: please move the common get connection logic in your adapter and prisma into a file in /libraries
> 
> It seems you also have a bug in your battr support; I think it is on imp.banner instead of imp
> 
> Please also document this endpoint change on the docs repo

Thanks for the comments; I've made the updates. How should we test this part? `Be sure to test the integration with your ad server using the [Hello World](https://github.com/integrationExamples/gpt/hello_world.html) sample page.` It seems the URL is not working.

**Review by patmmccann — APPROVED** [review]

**Comment by patmmccann** [comment]
This looks good to me, if you pull in master you'll pass that viant test. I noticed you marked as draft so holding off on merge

Thanks!

File: libraries/connectionInfo/connectionUtils.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+export function getConnectionType() {
+  const connection = navigator.connection || navigator.webkitConnection;
+  if (!connection) {
+    return 0;
+  }
+  switch (connection.type) {
+    case 'ethernet':
+      return 1;
+    case 'wifi':
```
Comment (Grammarian) [inline libraries/connectionInfo/connectionUtils.js:14]: Do we want to include "wimax" with this case?

**Comment by patmmccann** [comment]
> > Hi @tongwu-sh ? It seems your adapter needs a tiny bit of additional maintenance: please move the common get connection logic in your adapter and prisma into a file in /libraries
> > It seems you also have a bug in your battr support; I think it is on imp.banner instead of imp
> > Please also document this endpoint change on the docs repo
> 
> Thanks for the comments; I've made the updates. How should we test this part? `Be sure to test the integration with your ad server using the [Hello World](https://github.com/integrationExamples/gpt/hello_world.html) sample page.` It seems the URL is not working.

Fixed, #12011

File: libraries/connectionInfo/connectionUtils.js (hunk 1 above)
Comment (tongwu-sh) [reply libraries/connectionInfo/connectionUtils.js:14]: Yes, added. And checked for chrome/edge looks like connection.type is not always there but connection.effectiveType has value. Updated to handle that case.

**Comment by tongwu-sh** [comment]
@patmmccann , we finished the test for this change, could you help merge it? Thanks!

**Comment by tongwu-sh** [comment]
Hello @patmmccann , we finished tests for this change and ready to merge, could you help take another look at this PR? Thanks!

**Review by patmmccann — APPROVED** [review]

**Pull request merged**